crates/node-builder/src/components/traits.rs

```diff
@@ -0,0 +1,145 @@
+//! Traits for the builder
+
+use crate::{
+    components::NodeComponents,
+    node::{FullNodeTypes, NodeTypes},
+    BuilderContext,
+};
+use reth_network::NetworkHandle;
+use reth_payload_builder::PayloadBuilderHandle;
+use reth_tasks::TaskExecutor;
+use reth_transaction_pool::TransactionPool;
+
+/// Encapsulates all types and components of the node.
+pub trait FullNodeComponents: FullNodeTypes + 'static {
+    /// The transaction pool of the node.
+    type Pool: TransactionPool;
+
+    /// Returns the transaction pool of the node.
+    fn pool(&self) -> &Self::Pool;
+
+    /// Returns the provider of the node.
+    fn provider(&self) -> &Self::Provider;
+
+    /// Returns the handle to the network
+    fn network(&self) -> &NetworkHandle;
+
+    /// Returns the handle to the payload builder service.
+    fn payload_builder(&self) -> &PayloadBuilderHandle<Self::Engine>;
+
+    /// Returns the task executor.
+    fn executor(&self) -> &TaskExecutor;
```

Comment:
suggestion: maybe make this more verbose, i.e. `task_executor` since it could be mistaken for the tx executor/block executor? which we might want to expose at some point

Reply:
good point, will do